Features/absorption chillers (#74)
oemof/oemof-thermal: docs/absorption_chillers.rst

    :math:`COP`               :py:obj:`COP`                                       Coefficient of performance
    ========================= =================================================== ===========

Usage
_____

The following example shows how to calculate the COP of a small absorption chiller.
The characteristic coefficients used in this examples belong to a 10 kW absorption chiller developed and
tested at the Technische Universität Berlin [1].

.. code-block:: python

    import oemof.thermal.absorption_heatpumps_and_chillers as abs_chiller

    # Characteristic temperature difference
    ddt = abs_chiller.calc_characteristic_temp(
        t_hot=[85],  # in °C
        t_cool=[26],  # in °C
        t_chill=[15],  # in °C
        coef_a=10,
        coef_e=2.5,
        method='kuehn_and_ziegler')

    # Cooling capacity
    Q_dots_evap = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=0.42,
        coef_r=0.9,
        method='kuehn_and_ziegler')

    # Driving heat
    Q_dots_gen = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=0.51,
        coef_r=2,
        method='kuehn_and_ziegler')

>     COPs = Q_dots_evap / Q_dots_gen

Reviewer: ```suggestion
    COPs = [Qevap / Qgen for Qgen, Qevap in zip(Q_dots_gen, Q_dots_evap)]
```

    chiller on the cooling water temperature.

You find the code that is behind Fig.2 in our examples:
https://github.com/oemof/oemof-thermal/tree/master/examples

You can run the calculations for any other absorption heat pump or chiller by
entering the specific parameters (a, e, s, r) belonging to that specific machine.
The specific parameters are determined by a numerical fit of the
four parameters with testing data or data from the fact sheet (technical
specifications from the manufacturer) if temperatures for at least two
points of operation are given.
You find detailed information in the referenced papers.

This package comes with characteristic parameters for five absorption chillers.
Four published by Puig-Arnavat et al. [2]: 'Rotartica', 'Safarik', 'Broad_01' and 'Broad_02' 
and one published by Kühn and Ziegler [1]: 'Kuehn'.
If you like to contribute parameters for other machines,
please feel free to contact us or to contribute directly via github.

To model one of the machines provided by this package you can adapt the code
above in the following way. You find information on the machines in [2].

.. code-block:: python

>     import oemof.thermal.absorption_heatpumps_and_chillers as abs_chiller
> 
>     filename_charpara = os.path.join(os.path.dirname(__file__), 'data/characteristic_parameters.csv')
>     charpara = pd.read_csv(filename_charpara)
> 
>     chiller_name = 'Kuehn'  # 'Rotartica', 'Safarik', 'Broad_01', 'Broad_02'
> 
>      # Characteristic temperature difference
>     ddt = abs_chiller.calc_characteristic_temp(
>         t_hot=[85],  # in °C
>         t_cool=[26],  # in °C
>         t_chill=[15],  # in °C
>         coef_a=charpara[(charpara['name'] == chiller_name)]['a'].values[0],
>         coef_e=charpara[(charpara['name'] == chiller_name)]['e'].values[0],
>         method='kuehn_and_ziegler')
> 
>     # Cooling capacity
>     Q_dots_evap = abs_chiller.calc_heat_flux(
>         ddts=ddt,
>         coef_s=charpara[(charpara['name'] == chiller_name)]['s_E'].values[0],
>         coef_r=charpara[(charpara['name'] == chiller_name)]['r_E'].values[0],
>         method='kuehn_and_ziegler')
> 
>     # Driving heat
>     Q_dots_gen = abs_chiller.calc_heat_flux(
>         ddts=ddt,
>         coef_s=charpara[(charpara['name'] == chiller_name)]['s_G'].values[0],
>         coef_r=charpara[(charpara['name'] == chiller_name)]['R_G'].values[0],
>         method='kuehn_and_ziegler')
> 
>     COPs = Q_dots_evap / Q_dots_gen

Reviewer: ```suggestion
    import oemof.thermal.absorption_heatpumps_and_chillers as abs_chiller
    import pandas as pd
    import os

    filename_charpara = os.path.join(os.path.dirname(__file__), 'data/characteristic_parameters.csv')
    charpara = pd.read_csv(filename_charpara)

    chiller_name = 'Kuehn'  # 'Rotartica', 'Safarik', 'Broad_01', 'Broad_02'

     # Characteristic temperature difference
    ddt = abs_chiller.calc_characteristic_temp(
        t_hot=[85],  # in °C
        t_cool=[26],  # in °C
        t_chill=[15],  # in °C
        coef_a=charpara[(charpara['name'] == chiller_name)]['a'].values[0],
        coef_e=charpara[(charpara['name'] == chiller_name)]['e'].values[0],
        method='kuehn_and_ziegler')

    # Cooling capacity
    Q_dots_evap = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=charpara[(charpara['name'] == chiller_name)]['s_E'].values[0],
        coef_r=charpara[(charpara['name'] == chiller_name)]['r_E'].values[0],
        method='kuehn_and_ziegler')

    # Driving heat
    Q_dots_gen = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=charpara[(charpara['name'] == chiller_name)]['s_G'].values[0],
        coef_r=charpara[(charpara['name'] == chiller_name)]['r_G'].values[0],
        method='kuehn_and_ziegler')

    COPs = [Qevap / Qgen for Qgen, Qevap in zip(Q_dots_gen, Q_dots_evap)]
```

.. _absorption_chillers_label:

~~~~~~~~~~~~~~~~~~~~~~~
Absorption chillers
~~~~~~~~~~~~~~~~~~~~~~~

Calculations for absorption chillers based on the characteristic equation method.

Scope
_____

This module was developed to provide cooling capacity and COP calculations
based on temperatures for energy system optimizations with oemof.solph.

Concept
_______

A characteristic equation model to describe the performance of
absorption chillers.

> .. figure:: _pics/abs_chiller_col_02.png

Reviewer: Is it possible to add a dot to heat flows in the schematic?

Author: Done!

    :width: 40 %
    :alt: abs_chiller_col_02.png
    :align: center
    :figclass: align-center

    Fig.1: Absorption cooling (or heating) process.

The cooling capacity (:math:`\dot{Q}_{E}`) is determined by a function of
the characteristic

> temperature difference (:math:`\Delta\Delta T`) that combines the external

Reviewer: ```suggestion
temperature difference (:math:`\Delta\Delta t`) that combines the external
```

I would rather replace uppercase letter 'T' with lowercase letter 't' since the uppercase letter stands for internal temperatures and might be misleading. The replacement would also be consistent with the literature we use. It concerns the formulas in the docs and the docs schematic.

Author: You are absolutly right. I'll adapt it.

    :math:`COP`               :py:obj:`COP`                                       Coefficient of performance
    ========================= =================================================== ===========

Usage
_____

The following example shows how to calculate the COP of a small absorption chiller.
The characteristic coefficients used in this examples belong to a 10 kW absorption chiller developed and
tested at the Technische Universität Berlin [1].

.. code-block:: python

    import oemof.thermal.absorption_heatpumps_and_chillers as abs_chiller

    # Characteristic temperature difference
    ddt = abs_chiller.calc_characteristic_temp(
        t_hot=[85],  # in °C
        t_cool=[26],  # in °C
        t_chill=[15],  # in °C
        coef_a=10,
        coef_e=2.5,
        method='kuehn_and_ziegler')

    # Cooling capacity
    Q_dots_evap = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=0.42,
        coef_r=0.9,
        method='kuehn_and_ziegler')

    # Driving heat
    Q_dots_gen = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=0.51,
        coef_r=2,
        method='kuehn_and_ziegler')

    COPs = Q_dots_evap / Q_dots_gen

> Fig.2 illustrates how the cooling capacity and the COP of an absorption

Reviewer: ```suggestion
Fig. 2 illustrates how the cooling capacity and the COP of an absorption
```

Author: I prefere to stick with _Fig.2_ or rename it _Figure 2_.
But I realized the other modules do not have captions for their figueres... Personally, I am a fan of captions but we should try to unify the appearance...

> chiller (here the 10kW absorption chiller mentioned above) depend on the

Reviewer: ```suggestion
chiller (here the 10 kW absorption chiller mentioned above) depend on the
```

> colling water temperature, i.e. the mean external fluid temperature at

Reviewer: ```suggestion
cooling water temperature, i.e. the mean external fluid temperature at
```

absorber and condenser.

.. figure:: _pics/cooling_capacity_over_cooling_water_temperature_Kuehn.png
    :width: 80 %
    :alt: cooling_capacity_over_cooling_water_temperature_Kuehn.png
    :align: center
    :figclass: align-center

>     Fig.2: Dependency of the cooling capacity and the COP of a 10kW absorption
>     chiller on the cooling water temperature.
> 
> You find the code that is behind Fig.2 in our examples:

Reviewer: ```suggestion
    Fig. 2: Dependency of the cooling capacity and the COP of a 10 kW absorption
    chiller on the cooling water temperature.

You find the code that is behind Fig. 2 in our examples:
```

https://github.com/oemof/oemof-thermal/tree/master/examples

You can run the calculations for any other absorption heat pump or chiller by
entering the specific parameters (a, e, s, r) belonging to that specific machine.
The specific parameters are determined by a numerical fit of the
four parameters with testing data or data from the fact sheet (technical
specifications from the manufacturer) if temperatures for at least two
points of operation are given.
You find detailed information in the referenced papers.

> This package comes with characteristic parameters for five absorption chillers
> published by Puig-Arnavat et al. [2]:
> 'Rotartica', 'Safarik', 'Broad_01', 'Broad_02', 'Kuehn'.

Reviewer: ```suggestion
This package comes with characteristic parameters for five absorption chillers.
Four published by Puig-Arnavat et al. [2]: 'Rotartica', 'Safarik', 'Broad_01' and 'Broad_02' 
and one published by Kühn and Ziegler [1]: 'Kuehn'.
```

Author: Thank you! :)

If you like to contribute parameters for other machines,
please feel free to contact us or to contribute directly via github.

To model one of the machines provided by this package you can adapt the code
above in the following way. You find information on the machines in [2].

> .. code-block:: python

Reviewer: I think this code-block needs to be updated? It has changed in the example.

Author: Sorry, I don't see it... what has changed? What do I need to update exactly?

Reviewer: Sorry for that unhandy comment. The code is not running due to just a few things:

- pandas and os imports are missing
- The parameter 'R_G' needs to be changed to 'r_G'
- Q_dots_evap / Q_dots_gen is not possible for list type

You can find my suggestion below.

     # Characteristic temperature difference
    ddt = abs_chiller.calc_characteristic_temp(
        t_hot=[85],  # in °C
        t_cool=[26],  # in °C
        t_chill=[15],  # in °C
        coef_a=charpara[(charpara['name'] == chiller_name)]['a'].values[0],
        coef_e=charpara[(charpara['name'] == chiller_name)]['e'].values[0],
        method='kuehn_and_ziegler')

    # Cooling capacity
    Q_dots_evap = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=charpara[(charpara['name'] == chiller_name)]['s_E'].values[0],
        coef_r=charpara[(charpara['name'] == chiller_name)]['r_E'].values[0],
        method='kuehn_and_ziegler')

    # Driving heat
    Q_dots_gen = abs_chiller.calc_heat_flux(
        ddts=ddt,
        coef_s=charpara[(charpara['name'] == chiller_name)]['s_G'].values[0],
        coef_r=charpara[(charpara['name'] == chiller_name)]['r_G'].values[0],
        method='kuehn_and_ziegler')

    COPs = [Qevap / Qgen for Qgen, Qevap in zip(Q_dots_gen, Q_dots_evap)]

You find information on the machines in [1], [2] and [3].
Please be aware that [2] introduces a slightly different approach
(using an improved characteristic equation with :math:`\Delta\Delta t''`

> instead of :math:`\Delta\Delta t'`)
> and therefore find different characteristic parameters that the ones from [1] we
> are using here.

Reviewer: ```suggestion
instead of :math:`\Delta\Delta t'`).
The characteristic parameters that we use are derived from [2] and therefore differ from those in [1].
```
